src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -211,7 +211,9 @@ export class MatFormField
   }
   set appearance(value: MatFormFieldAppearance) {
     const oldValue = this._appearance;
-    this._appearance = value || this._defaults?.appearance || DEFAULT_APPEARANCE;
+    this._appearance = ['fill', 'outline'].includes(value)
```

Comment:
Someone in google was setting it to "standard" which resulted in a form-field that looked almost like a fill one but with some subtle bugs. I agree that the type checking should catch it. I'm not sure if that team had template type-checking off or just casted their value or what, but I figure its worth a check to avoid having to debug this confusing case again in the future. And yeah I can optimize the check

Reply:
Fair enough. Ideally we would limit it as `ngDevMode` only for it to be elided by Esbuild/terser, but it's a low resource check

Reply:
Done, PTAL